fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -149,17 +152,22 @@ where
     async fn deliver(
         &self,
         mut state: Self::State,
-        msg: Self::Message,
+        mut msg: Self::Message,
     ) -> anyhow::Result<(Self::State, Self::DeliverOutput)> {
         let (apply_ret, emitters, latency) = if msg.from == system::SYSTEM_ACTOR_ADDR {
             let (execution_result, latency) = measure_time(|| state.execute_implicit(msg.clone()));
             let (apply_ret, emitters) = execution_result?;
 
             (apply_ret, emitters, latency)
         } else {
+            // TODO: maybe compare the gas limits is better?
+            msg.gas_limit = msg.gas_limit.min(self.gas.available_block_gas());
+
```

Comment:
This is better modelled as a `pre_exec` and a `post_exec` hook. If the gas limit exceeds the available gas, we should abort, not cap the execution gas. It is the block proposer's mistake, not the user's mistake (if you do this, you'd likely provoke an out of gas despite them having set the gas limit correctly).